Zend/zend_inheritance.c

```diff
@@ -482,12 +482,23 @@ static inheritance_status zend_perform_covariant_type_check(
 			/* Replacing type that accepts self with static is okay */
 			added_types &= ~MAY_BE_STATIC;
 		}
+
+		if (added_types & MAY_BE_NORETURN) {
```

Comment:
This should probably check that the return type can *only* be `MAY_BE_NORETURN`. While `T|noreturn` can't be expressed I'm not sure if it can't be inferred.